File: 3541-find-most-frequent-vowel-and-consonant/3541-find-most-frequent-vowel-and-consonant.js
/**
 * @param {string} s
 * @return {number}
 */
var maxFreqSum = function(s) {
    let vowels=new Set("aeiou")
    let vowelCount=new Map()
    let consonantCount=new Map()
    for(let j of s){
        if(vowels.has(j)){
            vowelCount.set(j,(vowelCount.get(j) || 0)+1)
        }
        else{
            consonantCount.set(j,(consonantCount.get(j) || 0)+1)
        }
    }
    let maxVowel=0
    for(let [key,value] of vowelCount){
        if(value>maxVowel){
            maxVowel=value
        }
    }
    let maxConsonant=0
    for(let [key,value] of consonantCount){
        if(value>maxConsonant){
            maxConsonant=value
        }
    }

    return maxVowel+maxConsonant
};

File: 3541-find-most-frequent-vowel-and-consonant/3541-find-most-frequent-vowel-and-consonant.cpp
class Solution {
public:
    int maxFreqSum(string s) {
        unordered_map <char,int> vowelCount;
        unordered_map <char,int> consonantCount;
        int maxVowel=0;
        int maxConsonant=0;
        for(char ch : s){
            if(ch=='a' || ch=='e' || ch=='i' || ch=='o' || ch=='u'){
                vowelCount[ch]++;
                if(vowelCount[ch]>maxVowel){
                    maxVowel=vowelCount[ch];
                }
            }
            else{
                consonantCount[ch]++;
                if(consonantCount[ch]>maxConsonant){
                    maxConsonant=consonantCount[ch];
                }
            }
        }
        return maxVowel+maxConsonant;
    }
};